clang/test/SemaSYCL/intel-fpga-no-global-work-offset.cpp

```diff
@@ -1,68 +1,61 @@
 
-    // CHECK: SYCLIntelNoGlobalWorkOffsetAttr{{.*}}
-    // CHECK-NEXT:  ConstantExpr {{.*}} 'int'
-    // CHECK-NEXT:  value: Int 42
-    // CHECK-NEXT:  IntegerLiteral{{.*}}42{{$}}
-    h.single_task<class test_kernel3>(
-        []() [[intel::no_global_work_offset(42)]]{});
+struct S {};
+void var() {
+  // expected-note@+1{{in instantiation of function template specialization 'func<S>' requested here}}
+  func<S>();
+}
+
+// Test that checks expression is not a constant expression.
+// expected-note@+1{{declared here}}
+int foo();
+// expected-error@+2{{expression is not an integral constant expression}}
+// expected-note@+1{{non-constexpr function 'foo' cannot be used in a constant expression}}
+[[intel::no_global_work_offset(foo() + 12)]] void func1();
+
+// Test that checks expression is a constant expression.
+constexpr int bar() { return 0; }
+[[intel::no_global_work_offset(bar() + 12)]] void func2(); // OK
+
+// No diagnostic is thrown since arguments match. Silently ignore duplicate attribute.
+[[intel::no_global_work_offset]] void func3();
+[[intel::no_global_work_offset(1)]] void func3() {} // OK
 
-    // CHECK: SYCLIntelNoGlobalWorkOffsetAttr{{.*}}
-    // CHECK-NEXT:  ConstantExpr {{.*}} 'int'
-    // CHECK-NEXT:  value: Int -1
-    // CHECK-NEXT: UnaryOperator{{.*}} 'int' prefix '-'
-    // CHECK-NEXT-NEXT: IntegerLiteral{{.*}}1{{$}}
-    h.single_task<class test_kernel4>(
-        []() [[intel::no_global_work_offset(-1)]]{});
+[[intel::no_global_work_offset(0)]] void func4(); // expected-note {{previous attribute is here}}
+[[intel::no_global_work_offset]] void func4();    // expected-warning{{attribute 'no_global_work_offset' is already applied with different arguments}}
 
-    // Ignore duplicate attribute.
-    h.single_task<class test_kernel5>(
-    // CHECK:       SYCLIntelNoGlobalWorkOffsetAttr {{.*}}
-    // CHECK-NEXT:  ConstantExpr {{.*}} 'int'
-    // CHECK-NEXT:  value: Int 1
-    // CHECK-NEXT:  IntegerLiteral{{.*}}1{{$}}
-        []() [[intel::no_global_work_offset,
-               intel::no_global_work_offset]]{}); // OK
+// No diagnostic is emitted because the arguments match.
+[[intel::no_global_work_offset(1)]] void func5();
+[[intel::no_global_work_offset(1)]] void func5() {} // OK
 
-    // expected-error@+2{{integral constant expression must have integral or unscoped enumeration type, not 'const char[4]'}}
-    h.single_task<class test_kernel6>(
-        []() [[intel::no_global_work_offset("foo")]]{});
+// Diagnostic is emitted because the arguments mismatch.
+[[intel::no_global_work_offset(0)]] void func6(); // expected-note {{previous attribute is here}}
+[[intel::no_global_work_offset(1)]] void func6(); // expected-warning{{attribute 'no_global_work_offset' is already applied with different arguments}}
 
-    h.single_task<class test_kernel7>([]() {
-      // expected-error@+1{{'no_global_work_offset' attribute only applies to functions}}
-      [[intel::no_global_work_offset(1)]] int a;
```

Comment:
Did we lose coverage for this?

Reply:
No, I moved this to [clang/test/SemaSYCL/intel-fpga-no-global-work-offset.cpp](https://github.com/intel/llvm/pull/6184/files#diff-0d420fe07fdb14fef73cb83c5ac0962db1d4e2dc5fca293220852776999d2fdd)

Reply:
Ah ok. I didn't see it when I looked earlier. Thanks!

Reply:
No problem. Thanks for reviews!